include/sampling/sampling.hpp

```diff
@@ -68,169 +60,157 @@ template <
+    typename NT,
+    typename Point
 >
 void gaussian_sampling(PointList &randPoints,
                        Polytope &P,
                        RandomNumberGenerator &rng,
                        const unsigned int &walk_len,
                        const unsigned int &rnum,
                        const NT &a,
-                       const Point &starting_point,
-                       unsigned int const& nburns)
-{
+                       const Point &starting_point, 
+                       unsigned int const& nburns) {
 
-    typedef typename WalkTypePolicy::template Walk
-            <
-                    Polytope,
-                    RandomNumberGenerator
-            > walk;
+    typename WalkTypePolicy::template Walk<Polytope, RandomNumberGenerator> walk(P, starting_point, a, rng);
 
-    //RandomNumberGenerator rng(P.dimension());
-    PushBackWalkPolicy push_back_policy;
-
-    Point p = starting_point;
-
-    typedef GaussianRandomPointGenerator <walk> RandomPointGenerator;
+    Point p = starting_point; 
     if (nburns > 0) {
-        RandomPointGenerator::apply(P, p, a, nburns, walk_len, randPoints,
-                                    push_back_policy, rng);
-        randPoints.clear();
+        for (unsigned int i = 0; i < nburns; ++i) {
+            walk.apply(P, p, a, walk_len, rng); 
+        }
     }
-    RandomPointGenerator::apply(P, p, a, rnum, walk_len, randPoints,
-                                push_back_policy, rng);
 
+    randPoints.clear(); 
 
+    for (unsigned int i = 0; i < rnum; ++i) {
+        walk.apply(P, p, a, walk_len, rng); 
+        randPoints.push_back(p); 
+    }
 }
 
 
 template <
-        typename PointList,
-        typename Polytope,
-        typename RandomNumberGenerator,
-        typename WalkTypePolicy,
-        typename NT,
-        typename Point
-        >
+    typename PointList,
+    typename Polytope,
+    typename RandomNumberGenerator,
+    typename WalkTypePolicy,
+    typename NT,
+    typename Point
+>
 void gaussian_sampling(PointList &randPoints,
                        Polytope &P,
                        RandomNumberGenerator &rng,
-                       WalkTypePolicy &WalkType,
+                       WalkTypePolicy &WalkType, 
                        const unsigned int &walk_len,
                        const unsigned int &rnum,
                        const NT &a,
                        const Point &starting_point,
-                       unsigned int const& nburns)
-{
+                       unsigned int const& nburns) {
 
-    typedef typename WalkTypePolicy::template Walk
-            <
-                    Polytope,
-                    RandomNumberGenerator
-            > walk;
-
-    //RandomNumberGenerator rng(P.dimension());
-    PushBackWalkPolicy push_back_policy;
+    
+    typename WalkTypePolicy::template Walk<Polytope, RandomNumberGenerator> walk(P, starting_point, a, rng);
 
-    Point p = starting_point;
+    Point p = starting_point; 
```

Comment:
please do not add trailing whitespaces (there are more cases like this in this PR)